src/_pytest/skipping.py

```diff
@@ -101,6 +101,10 @@ def evaluate_condition(item: Item, mark: Mark, condition: object) -> Tuple[bool,
             "platform": platform,
             "config": item.config,
         }
+        for dictionary in item.config.hook.pytest_skipif_additional_globals():
+            if not isinstance(dictionary, dict) or not dictionary:
+                continue
+            globals_.update(dictionary)
```

Comment:
There may be need for a warning in case of conflicts

Reply:
So we allow overriding the `globals_` keys, but we should warn, correct?
Is there a good example on how to raise that warning?

And to be clear, we should not error, just warn.

Reply:
I would vote to do not warn or error out. Most users expect that something updating a dictionary will overwrite what's in there.

Let's see how this plays out first, adding a warning is simple to do later if we get reports of hard to tracking issues.